src/coreclr/tools/Common/Compiler/HardwareIntrinsicHelpers.cs

```diff
@@ -269,11 +229,16 @@ private static class Arm64IntrinsicConstants
             public const int Rcpc = 0x0100;
             public const int Rcpc2 = 0x0200;
             public const int Sve = 0x0400;
+            public const int Sve2 = 0x0800;
 
             public static void AddToBuilder(InstructionSetSupportBuilder builder, int flags)
             {
                 if ((flags & AdvSimd) != 0)
+                {
+                    // We need to explicitly track AdvSimd as some Linux machines can
+                    // still be encountered without it and we need to fail to launch
                     builder.AddSupportedInstructionSet("neon");
+                }
```

Comment:
It'd be nice if we could handle validation that baseline ISAs are supported more directly, that way we don't need to track them.

Maybe we could update `minipal_getcpufeatures` to return an additional value that indicates if the baseline is unsupported and then simply have `codeman` and the `nativeaot/runtime/startup.cpp` to error if these aren't there.

Reply:
We want the native code to be compiled with the same default baseline as the managed code default baseline. We do not want to leave perf of on the table for native code if we raise the default baseline.

If we wanted to produce nice error message for ancient hardware, it would have to be a very special piece of code that is compiled with lower baseline and that runs before anything else in the binary (e.g. that runs even before any static constructors). I would not bother. It is close to impossible to test that it works if anything else.

BTW: When the Windows C++ x86 CRT baseline got raised to SSE2, they did not bother to produce nice error message either (you would get very weird crash if you tried to run on non-SSE2 machine).

Reply:
Sounds like we shouldn't need to track AdvSimd and error like we are today then. I'll plan on cleaning that up in a follow up PR then, so I don't need to churn this one even more.